cpp/mrc/include/mrc/node/rx_node.hpp

```diff
@@ -234,9 +234,10 @@ class RxNodeComponent : public WritableProvider<InputT>, public WritableAcceptor
                 // Forward to the writable edge
                 this->get_writable_edge()->await_write(std::move(message));
             },
-            [this]() {
+            [this](std::exception_ptr ptr) {
                 // On completion, release connections
                 WritableAcceptor<OutputT>::release_edge_connection();
+                runnable::Context::get_runtime_context().set_exception(std::move(ptr));
             }));
```

Comment:
```suggestion
            [this](std::exception_ptr ptr) {
                runnable::Context::get_runtime_context().set_exception(std::move(ptr));
            },
            [this]() {
                // On completion, release connections
                WritableAcceptor<OutputT>::release_edge_connection();
            }));
```

Reply:
You still want the edge to be released on completion. Need to add the error handler in addition to the completion handler